### ISSUE TITLE
Kernel types sizing mistakes when using sizeof with CO-RE

### ISSUE THREAD

**Issue by yanivagman** [aquasecurity/tracee#1193]
We should be careful when using the sizeof operator on structs as our new slim version of vmlinux.h does not always contain all the fields of the struct.
As an example, sockaddr_in didn't contain the padding field (fixed in https://github.com/aquasecurity/tracee/pull/1190).
Another example is any place where we use READ_KERN, which uses sizeof to read the correct number of bytes of the requested field.

After going over our bpf code, I found two more structs that have incomplete definition but which are used with sizeof:
`struct tcphdr` and `struct udphdr`.

Other structs seem to have the complete definition.

To deal with such a problem, there are two helpers we can use (see "Sizing kernel types and fields" in https://nakryiko.com/posts/bpf-core-reference-guide/): `bpf_core_type_size()` and `bpf_core_field_size()`.
We might need to update our READ_KERN core macro to use `bpf_core_type_size()` to avoid future bugs.

Also, quoting from the bpf-core-reference-guide: "The general recommendation is to not read struct fields as a whole, if at all possible. Prefer reading just the primitive fields you are ultimately interested in."

In summary:
1. Add full definition of `struct tcphdr` and `struct udphdr`.
2. Update READ_KERN co-re macro to use  `bpf_core_type_size()` instead of `sizeof`
3. If possible, read the primitive fields instead of the full struct. Currently, we read the following structs in our bpf code:
    - path
    - iovec
    - sockaddr_un
    - sockaddr_in
    - sockaddr_in6
    - qstr
    - timespec64
    - in6_addr

**Comment by yanivagman** [comment]
To deal with such problem, there are two helpers we can use (see "Sizing kernel types and fields" in https://nakryiko.com/posts/bpf-core-reference-guide/): `bpf_core_type_size()` and `bpf_core_field_size()`.

Also, quoting from the bpf-core-reference-guide: "The general recommendation is to not read struct fields as a whole, if at all possible. Prefer reading just the primitive fields you are ultimately interested in."

**Comment by rafaeldtinoco** [comment]
I believe I have already addressed this with the new network code. The protocol headers are fully declared due to headers copying from skb into them (allocated in stack).
